client/app/views/popover_screens/guests.coffee

```diff
@@ -145,10 +163,17 @@ module.exports = class GuestPopoverScreen extends EventPopoverScreenView
         # view to be refreshed
         guests = _.clone guests
         guest  = guests[index]
-        # We add the information regarding the cozy: we change the label field
-        # so that the user has a visual feedback
-        guest.shareWithCozy = false
-        guest.label         = guest.email
+        # We remove the guest from the list to find a possible duplicate.
+        guests.splice index, 1
+
+        # Check for duplicate, same as above: if a guest already has that email
+        # then we check if `shareWithCozy` is true, if that's the case we can
+        # add the new guest; otherwise we cannot.
+        guestBis = _.findWhere(guests, email: guest.email)
+        if (not guestBis?) or (guestBis.shareWithCozy)
+            guest.shareWithCozy = false
+            guest.label         = guest.email
+            guests.splice index, 0, guest
 
```

Comment:
I see a pattern here, with the code above related to cozy duplicate. Do you think it's possible to mutualize somehow in a function ?